clientv3/integration/network_partition_test.go

```diff
@@ -40,35 +40,88 @@ func TestBalancerUnderNetworkPartitionPut(t *testing.T) {
 			return errExpected
 		}
 		return err
-	})
+	}, time.Second, true)
+}
+
+func TestBalancerUnderNetworkPartitionDelete(t *testing.T) {
+	testBalancerUnderNetworkPartition(t, func(cli *clientv3.Client, ctx context.Context) error {
+		_, err := cli.Delete(ctx, "a")
+		if err == context.DeadlineExceeded || err == rpctypes.ErrTimeout {
+			return errExpected
+		}
+		return err
+	}, time.Second, true)
+}
+
+func TestBalancerUnderNetworkPartitionTxn(t *testing.T) {
+	testBalancerUnderNetworkPartition(t, func(cli *clientv3.Client, ctx context.Context) error {
+		_, err := cli.Txn(ctx).
+			If(clientv3.Compare(clientv3.Version("foo"), "=", 0)).
+			Then(clientv3.OpPut("foo", "bar")).
+			Else(clientv3.OpPut("foo", "baz")).Commit()
+		if err == context.DeadlineExceeded || err == rpctypes.ErrTimeout {
+			return errExpected
+		}
+		return err
+	}, time.Second, true)
+}
+
+// TestBalancerUnderNetworkPartitionLinearizableGet tests when one member becomes isolated,
+// first Get request succeeds by switching endpoints within timeout.
+func TestBalancerUnderNetworkPartitionLinearizableGet(t *testing.T) {
+	testBalancerUnderNetworkPartition(t, func(cli *clientv3.Client, ctx context.Context) error {
+		_, err := cli.Get(ctx, "a")
+		return err
+	}, 7*time.Second, false)
 }
 
-// TestBalancerUnderNetworkPartitionGet tests when one member becomes isolated,
+// TestBalancerUnderNetworkPartitionLinearizableGetRetry tests when one member becomes isolated,
 // first Get request fails, and following retry succeeds with client balancer
 // switching to others.
-func TestBalancerUnderNetworkPartitionGet(t *testing.T) {
+func TestBalancerUnderNetworkPartitionLinearizableGetRetry(t *testing.T) {
 	testBalancerUnderNetworkPartition(t, func(cli *clientv3.Client, ctx context.Context) error {
 		_, err := cli.Get(ctx, "a")
 		if err == context.DeadlineExceeded {
 			return errExpected
 		}
 		return err
-	})
+	}, time.Second, true)
+}
+
+func TestBalancerUnderNetworkPartitionSerializableGet(t *testing.T) {
+	testBalancerUnderNetworkPartition(t, func(cli *clientv3.Client, ctx context.Context) error {
+		_, err := cli.Get(ctx, "a", clientv3.WithSerializable())
+		if err == context.DeadlineExceeded {
+			return errExpected
+		}
+		return err
+	}, 2*time.Second, false)
+}
+
+func TestBalancerUnderNetworkPartitionSerializableGetRetry(t *testing.T) {
```

Comment:
what is the difference between Get and GetRetry?

Reply:
`LinearizableGetWithRetry` expects quorum get succeeds on the second try after network partition happens.

`LinearizableGetWithNoRetry` expects quorum get blocks until balancer switch after network partition. Otherwise error out (without manual retry--expects first request succeeds).